src/components/RatingStar/RatingStar.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStar } from '@fortawesome/free-solid-svg-icons';
+
+const RatingStar = ({ starNum }) => {
+  const rating = starNum;
+  const ratingArr = new Array(5).fill(1);
+
+  return (
+    <span>
+      {ratingArr.map((num, idx) => {
+        return (
+          <FontAwesomeIcon
+            key={idx}
+            icon={faStar}
+            style={
+              num + idx <= rating ? { color: '#fa722d' } : { color: '#dadfe3' }
+            }
+          />
+        );
+      })}
+    </span>
+  );
+};
```

Comment:
```suggestion
const RatingStar = ({ rating, setStarNum, setHover, type }) => {
  const ratingArr = new Array(5).fill(1);

  const clickRatingNum = num => {
    type === 'select' && setStarNum(num);
  };

  const hoverRating = num => {
    type === 'select' && setHover(num);
  };

  return (
    <span>
      {ratingArr.map((num, idx) => {
        return (
          <FontAwesomeIcon
            key={idx}
            icon={faStar}
            style={
              num + idx <= rating ? { color: '#fa722d' } : { color: '#dadfe3' }
            }
            onClick={() => clickRatingNum(num + idx)}
            onMouseEnter={() => hoverRating(num + idx)}
            onMouseLeave={() => hoverRating(0)}
          />
        );
      })}
    </span>
  );
};
```

해당 코드 이렇게 수정하면 별 컴포넌트 하나로 별점과 별점 표시하는 모든 기능을 커버 가능합니다!
하지만 왜 이렇게 해야 하는지는 유정님이 코드 보시면서 분석해 보시고
이전 코드는 어떠했는데, 하나의 컴포넌트로 별점 관련 기능을 구현하기 위해서
이러한 코드로 적용했다 라고 하는 주제로 블로그 글을 작성해주세요!

틀은 [노션](https://www.notion.so/wecode/useEffect-b9c095cf38244e6a910d12f70d36740f) 첨부 드립니다!